pkg/infoschema/builder.go

```diff
@@ -970,9 +971,10 @@ func NewBuilder(r autoid.Requirement, factory func() (pools.Resource, error), in
 }
 
 func tableBucketIdx(tableID int64) int {
+	intest.Assert(tableID > 0)
```

Comment:
Will this `tableID > 0` be false in the test scenario? If so, whether it is possible to encounter real scenes. If not, is the test unnecessary?

Reply:
We found `tableID <= 0` in some test cases but they are handled before `tableBucketIdx`. Adding assert here for two propose:

- Provide a more clear reason for the failed test even if it would also fail without the assert function. The assert function is a more straightforward way to check the constraints.

- This assert also checks `tableID` should not be `0`. It may not fail if some codes building infoschema with an zero id table.